src/webgl/ShaderGenerator.js

```diff
@@ -208,10 +206,10 @@ function shadergenerator(p5, fn) {
       if (this.swizzleChanged) { return true; }
       if (this.isInternal || isVariableNode(this)) { return false; }
       let score = 0;
-      score += isBinaryOperatorNode(this);
+      score += isBinaryOperatorNode(this) * 2;
       score += isVectorNode(this) * 2;
-      score += this.usedIn.length;
-      return score > 3;
+      score += Math.max(0, this.usedIn.length - 1);
```

Comment:
Just a thought since we're adjusting the score here: I think function call and component nodes currently don't set `usedIn` for their children, if we add that does that change the score logic here?